core/commands/log_graph.py

```diff
@@ -152,81 +152,59 @@ def get_graph_args(self):
         return args
 
 
-class GsLogGraphCommand(WindowCommand, GitCommand):
-    def run(self, file_path=None, current_file=False):
-        self._file_path = self.file_path if current_file else file_path
-        options_array = [
-            "For current branch",
-            "For all branches",
-            "Filtered by author",
-            "Filtered by branch",
-        ]
-        self.window.show_quick_panel(
-            options_array,
-            self.on_option_selection,
-            flags=sublime.MONOSPACE_FONT
-        )
-
-    def on_option_selection(self, index):
-        if index == -1:
-            return
-
-        if index == 0:
-            self.window.run_command("gs_log_graph_current_branch", {"file_path": self._file_path})
-        elif index == 1:
-            self.window.run_command("gs_log_graph_current_branch", {"file_path": self._file_path})
-        elif index == 2:
-            self.window.run_command("gs_log_graph_by_author", {"file_path": self._file_path})
-        elif index == 3:
-            self.window.run_command("gs_log_graph_by_branch", {"file_path": self._file_path})
+class GsLogGraphCommand(GsLogCommand):
+    default_actions = [
+        ["gs_log_graph_current_branch", "For current branch"],
+        ["gs_log_graph_all_branches", "For all branches"],
+        ["gs_log_graph_by_author", "Filtered by author"],
+        ["gs_log_graph_by_branch", "Filtered by branch"],
+    ]
 
 
 class GsLogGraphActionCommand(GsLogActionCommand):
 
     """
     Checkout the commit at the selected line. It is also used by compare_commit_view.
     """
-
-    def run(self):
-        self.actions = [
-            ["show_commit", "Show commit"],
-            ["checkout_commit", "Checkout commit"],
-            ["compare_against", "Compare commit against ..."],
-            ["copy_sha", "Copy the full SHA"]
-        ]
-
+    default_actions = [
+        ["show_commit", "Show commit"],
+        ["checkout_commit", "Checkout commit"],
+        ["compare_against", "Compare commit against ..."],
+        ["copy_sha", "Copy the full SHA"]
+    ]
+
+    def update_actions(self):
+        super().update_actions()
         view = self.window.active_view()
         if view.settings().get("git_savvy.compare_commit_view.target_commit") == "HEAD":
             self.actions.append(["cherry_pick", "Cherry-pick commit"])
 
         if view.settings().get("git_savvy.log_graph_view"):
-            self.actions = self.actions + [
+            self.actions.extend([
                 ["diff_commit", "Diff commit"],
-                ["diff_commit_cache", "Diff commit (cached)"]
-            ]
+                ["diff_commit_cache", "Diff commit (cached)"],
+            ])
 
-        self._file_path = self.file_path
         if self._file_path:
             self.actions.insert(1, ["show_file_at_commit", "Show file at commit"])
```

Comment:
Sort of, I just forgot to pass the evaluated `self.file_path` property to the parent class, so it was using default `None` .. should be fixed now